ui/src/app/views/cohort-list/component.ts

```diff
@@ -66,6 +71,13 @@ export class CohortListComponent implements OnInit, OnDestroy {
     }
   }
 
+  delete(cohort: Cohort): void {
+    this.cohortsService.deleteCohort(this.wsNamespace, this.wsId, cohort.id).subscribe(() => {
+      this.cohortList = this.cohortList.splice(
+        this.cohortList.findIndex((index) => index === cohort) - 1, 1);
```

Comment:
This splice indexing looks wrong.

1. splice returns the spliced out elements, and mutates the original value.
2. -1 on the index doesn't make sense, since this would splice at -1 (i.e. the last element) when you delete the first cohort.

Test?

Reply:
Yeah. This was interesting, because I noticed it was actually off one the wrong direction, so this was a fix for that. Let me see if indexOf is better...

Reply:
Could you add a unit test covering this? I realize there's no test file for this component yet, but would be great if you're willing to take the time to add one.

Reply:
Can do!